src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +79,42 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /**
+   * Gets messages posted by multiple users
+   *
+   * @return a list of messages posted by multiple users. This list is the public feed
+   */
+  public List<Message> getAllMessages(){
+
+    // Initialize a list to store all messages
+    List<Message> messages = new ArrayList<>();
+  
+    // Query the database for a list of Entities
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+  
+    // For each of the entities (user messages), grab all info and create a "Message" object
+    for (Entity entity : results.asIterable()) {
+     try {
+      String idString = entity.getKey().getName();
+      UUID id = UUID.fromString(idString);
+      String user = (String) entity.getProperty("user");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+  
+      // A Message object is constructed from a user's UUID, user's email, their message, and a timestamp
+      Message message = new Message(id, user, text, timestamp);
+      messages.add(message);
+      
+     } catch (Exception e) {
+      System.err.println("Error reading message.");
+      System.err.println(entity.toString());
+      e.printStackTrace();
+     }
```

Comment:
Please make the indentation of this block consistent with the rest of the file. In particular, this file uses an indentation width of 2 spaces, but these for-loop and try-catch blocks are indented with one space.

Reply:
Fixed!